src/Components/Notifications/UnseenNotificationsIndicator.tsx

```diff
@@ -0,0 +1,80 @@
+import { createContext } from "react"
+import { DateTime } from "luxon"
+import { useState, useEffect, FC, useContext } from "react"
+
+export const LAST_SEEN_NOTIFICATION_PUBLISHED_AT_KEY =
+  "last_seen_notification_published_at"
+
+interface UnseenNotificationsIndicatorContextProps {
+  hasUnseenNotifications: boolean
+  setLastSeenNotificationDatetime: (datetime: string) => void
+}
+
+const UnseenNotificationsIndicatorContext = createContext<
+  UnseenNotificationsIndicatorContextProps
+>({
+  hasUnseenNotifications: false,
+  setLastSeenNotificationDatetime: () => {},
+})
+
+export const useUnseenNotificationsContext = () => {
+  return useContext(UnseenNotificationsIndicatorContext)
+}
+
+interface ProviderProps {
+  lastSeenNotificationDatetime: string
+}
+
+export const UnseenNotificationsIndicatorContextProvider: FC<ProviderProps> = ({
+  children,
+  lastSeenNotificationDatetime,
+}) => {
+  const [hasUnseenNotifications, setHasUnseenNotifications] = useState(false)
+
+  useEffect(() => {
+    setHasUnseenNotifications(
+      checkUnseenNotifications(lastSeenNotificationDatetime)
+    )
+  }, [lastSeenNotificationDatetime])
+
+  const setLastSeenNotificationDatetime = (
+    lastNotificationPublishedAt: string
+  ) => {
+    window.localStorage.setItem(
+      LAST_SEEN_NOTIFICATION_PUBLISHED_AT_KEY,
+      lastNotificationPublishedAt
+    )
+    setHasUnseenNotifications(false)
+  }
+
+  const providerValues: UnseenNotificationsIndicatorContextProps = {
+    hasUnseenNotifications,
+    setLastSeenNotificationDatetime,
+  }
+
+  return (
+    <UnseenNotificationsIndicatorContext.Provider value={providerValues}>
+      {children}
+    </UnseenNotificationsIndicatorContext.Provider>
+  )
+}
+
+// hasUnseenNotifications
+const checkUnseenNotifications = (lastNotificationDatetime: string) => {
+  if (!lastNotificationDatetime) {
+    return false
+  }
+
+  const prevLastNotificationDatetime = window.localStorage.getItem(
+    LAST_SEEN_NOTIFICATION_PUBLISHED_AT_KEY
+  )
+
+  if (!prevLastNotificationDatetime) {
+    return true
+  }
+
+  const prevDate = DateTime.fromISO(prevLastNotificationDatetime)
+  const newDate = DateTime.fromISO(lastNotificationDatetime)
+
+  return newDate > prevDate
+}
```

Comment:
Why not just have this hook execute the query for the data it needs? (We can support Relay hooks)